File: backend/app/schemas/notification.py
from datetime import datetime

from app.schemas.common import ORMModel


class NotificationResponse(ORMModel):
    id: int
    recipient_id: int
    event_type: str
    title: str
    content: str
    level: str
    related_type: str | None
    related_id: str | None
    delivered_channels: list | None
    status: str
    read_at: datetime | None
    created_at: datetime

File: frontend/src/types/notification.ts
export interface AppNotification {
  id: number
  recipient_id: number
  event_type: string
  title: string
  content: string
  level: string
  related_type?: string
  related_id?: string
  delivered_channels?: string[]
  status: 'unread' | 'read'
  read_at?: string
  created_at: string
}
